// xdb-digitalbits-sdk/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.version = void 0;
var tslib_1 = require("tslib");
require("es6-promise").polyfill();
var version = require("../package.json").version;
exports.version = version;
tslib_1.__exportStar(require("./frontier_api"), exports);
tslib_1.__exportStar(require("./server_api"), exports);
tslib_1.__exportStar(require("./account_response"), exports);
tslib_1.__exportStar(require("./errors"), exports);
var config_1 = require("./config");
Object.defineProperty(exports, "Config", { enumerable: true, get: function () { return config_1.Config; } });
var server_1 = require("./server");
Object.defineProperty(exports, "Server", { enumerable: true, get: function () { return server_1.Server; } });
var federation_server_1 = require("./federation_server");
Object.defineProperty(exports, "FederationServer", { enumerable: true, get: function () { return federation_server_1.FederationServer; } });
Object.defineProperty(exports, "FEDERATION_RESPONSE_MAX_SIZE", { enumerable: true, get: function () { return federation_server_1.FEDERATION_RESPONSE_MAX_SIZE; } });
var digitalbits_toml_resolver_1 = require("./digitalbits_toml_resolver");
Object.defineProperty(exports, "DigitalBitsTomlResolver", { enumerable: true, get: function () { return digitalbits_toml_resolver_1.DigitalBitsTomlResolver; } });
Object.defineProperty(exports, "DIGITALBITS_TOML_MAX_SIZE", { enumerable: true, get: function () { return digitalbits_toml_resolver_1.DIGITALBITS_TOML_MAX_SIZE; } });
var frontier_axios_client_1 = require("./frontier_axios_client");
Object.defineProperty(exports, "FrontierAxiosClient", { enumerable: true, get: function () { return frontier_axios_client_1.default; } });
Object.defineProperty(exports, "SERVER_TIME_MAP", { enumerable: true, get: function () { return frontier_axios_client_1.SERVER_TIME_MAP; } });
Object.defineProperty(exports, "getCurrentServerTime", { enumerable: true, get: function () { return frontier_axios_client_1.getCurrentServerTime; } });
tslib_1.__exportStar(require("./utils"), exports);
tslib_1.__exportStar(require("xdb-digitalbits-base"), exports);
exports.default = module.exports;
